springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,420 @@
+      status = HttpStatus.BAD_REQUEST;
+      messageBuilder.append(
+          "Bad request. Please check query parameters and execution logs for more information.\n");
+      return generateResponse(messageBuilder, status);
+    }
+    ValidatorLoader validatorLoader;
+    GtfsInput gtfsInput;
+    GtfsFeedContainer feedContainer;
+    NoticeContainer noticeContainer = new NoticeContainer();
+    try {
+      validatorLoader = new ValidatorLoader();
+    } catch (ValidatorLoaderException e) {
+      return generateResponse(
+          messageBuilder.append(e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+    GtfsFeedLoader feedLoader = new GtfsFeedLoader();
+    try {
+      gtfsInput = createGtfsInput(args);
+    } catch (IOException | URISyntaxException ioException) {
+      logger.atSevere().withCause(ioException);
+      return generateResponseAndExportReport(
+          messageBuilder.append(ioException.getMessage()),
+          HttpStatus.BAD_REQUEST,
+          noticeContainer,
+          args);
+    }
+    ValidationContext validationContext =
+        ValidationContext.builder()
+            .setCountryCode(
+                CountryCode.forStringOrUnknown(
+                    args.getCountryCode() == null ? CountryCode.ZZ : args.getCountryCode()))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+
+    try {
+      feedContainer =
+          Main.loadAndValidate(
+              validatorLoader, feedLoader, noticeContainer, gtfsInput, validationContext);
+    } catch (InterruptedException e) {
+      logger.atSevere().withCause(e).log("Validation was interrupted");
+      messageBuilder.append("Internal error. Please execution logs for more information.\n");
+      return generateResponse(
+          messageBuilder.append(e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+    Main.closeGtfsInput(gtfsInput, noticeContainer);
+    printSummary(startNanos, feedContainer);
+    messageBuilder.append("Execution of the validator was successful.\n");
+    exportReport(noticeContainer, args);
+    return generateResponse(
+        messageBuilder,
+        pushValidationReportToCloudStorage(commit_sha, dataset_id, args, messageBuilder));
+  }
+
+  /**
+   * Converts query parameters to {@code Argument}.
+   *
+   * @param output_base Base directory to store the outputs
+   * @param threads Number of threads to use
+   * @param country_code Country code of the feed, e.g., `nl`. It must be a two-letter country code
+   *     (ISO 3166-1 alpha-2)").
+   * @param url The fully qualified URL to download GTFS archive
+   * @param validation_report_name The name of the validation report including .json extension.
+   * @param system_error_report_name The name of the system error report including .json extension.
+   * @return the {@code Argument} instance generated from the query parameters passed.
+   */
+  private Arguments queryParametersToArguments(
```

Comment:
In this method signature and in `run()` pipe variables are using snake case (with `_` - `output_base`) instead of the normal Java convention camel case (`outputBase`). Seems like at a minimum this method should be changed to use camel case, and `run()` as well unless Spring Boot requires snake case for some reason.

Reply:
At first I thought query parameters had to be in snake case by convention. However, checking it seems that camelCase is also accepted. https://github.com/MobilityData/gtfs-validator/pull/933/commits/3bfbdda1e34f02b5979c7884920ae6ba3ad85aa5 changes snake case to camel case.

https://api.gov.au/standards/national_api_standards/naming-conventions.html